src/Workspaces/Core/Portable/Shared/Extensions/DocumentExtensions.cs

```diff
@@ -176,17 +176,17 @@ public static Task<SyntaxTreeIndex> GetSyntaxTreeIndexAsync(this Document docume
         /// Returns the semantic model for this document that may be produced from partial semantics. The semantic model
         /// is only guaranteed to contain the syntax tree for <paramref name="document"/> and nothing else.
         /// </summary>
-        public static async Task<SemanticModel> GetPartialSemanticModelAsync(this Document document, CancellationToken cancellationToken)
+        public static Task<SemanticModel> GetPartialSemanticModelAsync(this Document document, CancellationToken cancellationToken)
         {
-            if (document.Project.TryGetCompilation(out var compilation))
+            if (document.Project.TryGetCompilation(out _))
```

Comment:
Hard to tell if this actually is losing some of the original intent of this function: if we did fetch a compilation, we'd want to make sure the compilation is held alive prior to calling document.GetSemanticModelAsync(). The function probably should still capture compilation, then do:

```
var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
GC.KeepAlive(compilation);
return semanticModel;
```

The lack of GC.KeepAlive here is probably bad always, but discarding the variable _might_ impact JIT and thus make things even worse.

Reply:
I verified that there is no IL codegen difference between these 3 variants (async + out var, out var, discard) - all discard the value, so the compilation wasn't kept alive before also. Should I restore the async state machine here and add a GC.KeepAlive instead then?